### PR TITLE
Expose method to pull just diagnostics

### PR DESCRIPTION
**Pull request by sisuresh**
### What

Required for https://github.com/stellar/rs-soroban-sdk/pull/1265.

### CODE REVIEW

File: soroban-env-host/src/host.rs [hunk 1]
```diff
@@ -618,6 +619,19 @@ impl Host {
         }
     }
 
+    #[cfg(any(test, feature = "testutils"))]
+    pub fn with_debug_mode<F>(&self, f: F)
```
Comment (jayz22) [inline soroban-env-host/src/host.rs:623]: I've commented on the other PR (sdk), not sure why this needs to be exposed.

Comment (leighmcculloch) [reply soroban-env-host/src/host.rs:623]: Is the only difference between these functions the `pub` vs `pub(crate)`?

Comment (sisuresh) [reply soroban-env-host/src/host.rs:623]: yeah

Comment (sisuresh) [reply soroban-env-host/src/host.rs]: I changed things around to just expose a method that pull diagnostics.

**Review by dmkozh — APPROVED** [review]

**Review by leighmcculloch — APPROVED** [review]

File: soroban-env-macros/src/call_macro_with_all_host_functions.rs [hunk 2]
```diff
@@ -93,9 +93,15 @@ pub fn generate(file_lit: LitStr) -> Result<TokenStream, Error> {
 
             let r#return = format_ident!("{}", &f.r#return);
 
-            quote! {
-                #[doc = #docs]
-                { #export, #min_proto, #max_proto, fn #name(#(#args),*) -> #r#return }
+            if docs.is_empty() {
```
Comment (leighmcculloch) [inline soroban-env-macros/src/call_macro_with_all_host_functions.rs:96]: Would also be fine to generate a compiler error if docs is empty with the `compiler_error!` macro, with a note telling the dev to add a doc. Host fns should have docs.

**Review by leighmcculloch — APPROVED** [review]
Lgtm

File: soroban-env-macros/src/call_macro_with_all_host_functions.rs (hunk 2 above)
Comment (sisuresh) [reply soroban-env-macros/src/call_macro_with_all_host_functions.rs:96]: Yeah that makes sense. I didn't want to update env.json in this PR so went with this instead. I can open an issue if you think it's worth it.

Comment (leighmcculloch) [reply soroban-env-macros/src/call_macro_with_all_host_functions.rs:96]: I think this is fine.

**Pull request merged**